app/uk/gov/hmrc/securemessage/models/core/FilterTag.scala

```diff
@@ -25,8 +25,11 @@ object FilterTag {
     Json.reads[FilterTag]
   }
 
-  def parse(tagString: String): FilterTag = {
+  def parse(tagString: String): Either[String, FilterTag] = {
     val tag = tagString.split('~')
-    FilterTag(tag.head, tag(1))
+    tag.size match {
+      case 2 => Right(FilterTag(tag.head, tag(1)))
```

Comment:
If we dont specify a tag we get all ?

Reply:
No this will only be called if we add a tag parameter. The binder is supposed to return a Right with the FilterTag if it can be parsed otherwise a Left with an error message. The old implementation would throw an exception if we passed something like this:

Reply:
So we only take first Tag if we have 2 ?

Reply:
We will take all tags and bind them to FilterTag and pass to the controller a Option[List[FilterTag]]